app/models/person_duplicate.rb

```diff
@@ -29,6 +29,8 @@ class PersonDuplicate < ActiveRecord::Base
 
   validates :person_1_id, uniqueness: {scope: [:person_2_id]}
 
+  validate :valid_role_operations, if: -> { validation_context == {context: :merge} }
```

Comment:
following the shortening above, we need to shorten things here as well...

```suggestion
  validate :valid_role_operations, if: -> { validation_context == :merge }
```